File: gcode/src/main/java/org/qw3rtrun/p3d/g/event/WaitReceived.java
package org.qw3rtrun.p3d.g.event;

import lombok.Getter;
import org.qw3rtrun.p3d.core.msg.WaitReceivedEvent;

public record WaitReceived() implements WaitReceivedEvent {

    private static final WaitReceived instance = new WaitReceived();

    public static WaitReceived getInstance() {
        return instance;
    }
}

File: gcode/src/main/kotlin/org/qw3rtrun/p3d/g/marlin/event/WaitReceived.kt
package org.qw3rtrun.p3d.g.marlin.event

import org.qw3rtrun.p3d.core.msg.WaitReceivedEvent

class WaitReceived : WaitReceivedEvent {
    override fun toString(): String = "WaitReceived()"

    override fun equals(other: Any?): Boolean = other is WaitReceived

    override fun hashCode(): Int = WaitReceived::class.hashCode()

    companion object {
        @JvmField
        val INSTANCE: WaitReceived = WaitReceived()

        @JvmStatic
        fun getInstance(): WaitReceived = INSTANCE
    }
}
